src/repr/src/datum_vec.rs

```diff
@@ -59,24 +59,25 @@ impl DatumVec {
         borrow
     }
 
-    /// Borrow an instance with a specific lifetime, and pre-populate with some `Datum`s and a `Row`,
+    /// Borrow an instance with a specific lifetime, and pre-populate with two iterators of`Datum`s,
     /// for example first adding a key followed by its values.
-    pub fn borrow_with_iter<'a, I, D, R>(
+    pub fn borrow_with_iter_pair<'a, I1, I2, D>(
         &'a mut self,
-        datum_iter: I,
-        row: &'a R,
+        datum_iter1: I1,
+        datum_iter2: I2,
     ) -> DatumVecBorrow<'a>
     where
-        I: IntoIterator<Item = D>,
+        I1: IntoIterator<Item = D>,
+        I2: IntoIterator<Item = D>,
         D: ::std::borrow::Borrow<Datum<'a>>,
-        R: ::std::borrow::Borrow<Row> + 'a,
     {
         let mut borrow = self.borrow();
-        for datum in datum_iter {
+        for datum in datum_iter1 {
+            borrow.push(*datum.borrow());
+        }
+        for datum in datum_iter2 {
             borrow.push(*datum.borrow());
         }
-        let row_iter = row.borrow().iter();
-        borrow.extend(row_iter);
         borrow
     }
```

Comment:
I have the feeling we should deprecate the `borrow_with_*` functions. It seems to be better to write:
```
let mut b = datum_vec.borrow();
b.extend(datum_iter1);
b.extend(datum_iter2);
```
In the past this was less of a problem because we just passed in a slice, so this function was variadic, but since we're seeing different types, this isn't possible anymore.

It's fine to leave it as out-of-scope for this PR, but it'd be great to file an issue so we pick this up later.

Reply:
Yeah, that would work. For the PR, I just eliminated `borrow_with_iter_pair` and used your suggestion already, since it was a self-contained change. But, I did not touch the rest of the code. Here is the issue so that we do not forget: https://github.com/MaterializeInc/materialize/issues/22522. I also added there the need to refactor transits through `Row` in a few places we still call `into_row`.